libs/pika/async_cuda/tests/unit/cuda_sender.cu

```diff
@@ -23,11 +23,15 @@ namespace cu = pika::cuda::experimental;
 namespace ex = pika::execution::experimental;
 namespace tt = pika::this_thread::experimental;
 
-constexpr auto cuda_launch_kernel = [](auto&&... ts) {
+constexpr auto cuda_launch_kernel =
+    [](auto&&... ts) -> decltype(cu::check_cuda_error(
+                         cudaLaunchKernel(std::forward<decltype(ts)>(ts)...))) {
     cu::check_cuda_error(cudaLaunchKernel(std::forward<decltype(ts)>(ts)...));
 };
 
-constexpr auto cuda_memcpy_async = [](auto&&... ts) {
+constexpr auto cuda_memcpy_async =
+    [](auto&&... ts) -> decltype(cu::check_cuda_error(
+                         cudaMemcpyAsync(std::forward<decltype(ts)>(ts)...))) {
```

Comment:
Note, these kind of helpers could be refactored into a common header. However, I think these are perfect candidates for a potential HIP/CUDA header-only compatibility library, so I'm not doing anything about them here.